filter/filter_test.go

```diff
@@ -17,3 +20,31 @@ func TestIsExcluded(t *testing.T) {
 		assert.False(t, f.IsExcluded("c"))
 	})
 }
+
+func TestTargetsTypesWithIDs(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		f := filter.Filter{Targets: []string{"aws_instance.2", "aws_instance.3", "aws_iam_user.2", "aws_instance.2"}}
+		assert.Equal(t, map[string][]string{
+			"aws_instance": []string{"2", "3"},
+			"aws_iam_user": []string{"2"},
+		}, f.TargetsTypesWithIDs())
+	})
```

Comment:
Maybe add a test with: invalid types / IDs with extra dots?

Reply:
Added one for the IDs dots.

Reply:
Added one for the IDs.